test/parallel/test-quic-http3-push.js

```diff
@@ -0,0 +1,145 @@
+// Flags: --expose-internals --no-warnings
+'use strict';
+
+// Tests a simple QUIC HTTP/3 client/server round-trip
+
+const common = require('../common');
+if (!common.hasQuic)
+  common.skip('missing quic');
+
+const Countdown = require('../common/countdown');
+const assert = require('assert');
+const fixtures = require('../common/fixtures');
+const key = fixtures.readKey('agent1-key.pem', 'binary');
+const cert = fixtures.readKey('agent1-cert.pem', 'binary');
+const ca = fixtures.readKey('ca1-cert.pem', 'binary');
+
+const { createSocket } = require('quic');
+
+let client;
+const server = createSocket();
+
+const countdown = new Countdown(2, () => {
+  server.close();
+  client.close();
+});
+
+server.listen({ key, cert, ca, alpn: 'h3-24' });
+
+server.on('session', common.mustCall((session) => {
+
+  session.on('stream', common.mustCall((stream) => {
+    assert(stream.submitInitialHeaders({ ':status': '200' }));
+
+    const push = stream.pushStream({
+      ':method': 'GET',
+      ':scheme': 'https',
+      ':authority': 'localhost',
+      ':path': '/foo'
+    });
+    assert(push);
+    push.submitInitialHeaders({ ':status': '200' });
+    // TODO(@jasnell): There's currently a bug if we only call end() to write.
```

Comment:
Just curious, what happens in that case?

Reply:
At the time there was a failed assertion. I need to retest.